Yoast/javascript
# Refactor the Sentence and SentencePart classes

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Refactors `Sentence` and `SentencePart` classes.

## Relevant technical choices:

* `SentencePart` is changed to `Clause`.
* Since `Participle` class will be deleted in the future, the checks that are currently done inside the class are integrated to the new `Clause` class

## Test instructions

This PR can be tested by following these steps:

* Run yarn test and make sure all tests pass.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-752
